File: src/main/java/com/box/Box.java
package com.box;

import java.util.ArrayList;


public abstract class Box {
    float length;
    float width;
    float height;

    public boolean validate(ArrayList<Float> sizeDate){
       boolean b =  validate(sizeDate.get(0),sizeDate.get(1),sizeDate.get(2));
       return b;
    }
    public boolean validate(float length,float width,float height){
        return length <= this.length && width <= this.width && height <= this.height;

    }
    public abstract String getBoxType();



}

File: src/main/kotlin/com/box/Box.kt
package hd.com.box

open  abstract class Box(var length: Float , var  width: Float, var height: Float){



    fun validate(sizeDate :List<Float>) :Boolean = validate(sizeDate.get(0), sizeDate.get(1), sizeDate.get(2))
    fun validate(length :Float , width :Float , height :Float ) : Boolean {
       val _length = if (length > width) length else width
        var _width = if( length > width) width else length
       return _length <= this.length && _width <= this.width && height <= this.height
    }
    abstract fun getBoxPrice() :Int

    abstract fun getBoxType() :String



}